src/Symfony/Component/Form/Extension/Core/Type/FileType.php

```diff
@@ -12,15 +12,40 @@
 namespace Symfony\Component\Form\Extension\Core\Type;
 
 use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Form\FormEvent;
+use Symfony\Component\Form\FormEvents;
 use Symfony\Component\Form\FormInterface;
 use Symfony\Component\Form\FormView;
+use Symfony\Component\OptionsResolver\Options;
 use Symfony\Component\OptionsResolver\OptionsResolver;
 
 class FileType extends AbstractType
 {
     /**
      * {@inheritdoc}
      */
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        if ($options['multiple']) {
+            $builder->addEventListener(FormEvents::PRE_SUBMIT, function (FormEvent $event) {
+                $form = $event->getForm();
+                $data = $event->getData();
+
+                // submitted data for an input file (not required) without choosing any file
+                if (array(null) === $data) {
```

Comment:
Still not sure about this, by default only 1 `<input>` is rendered, but it can in fact be many (ie. created by the client with a "Add file" button), hence `multiple`.

So if many inputs are submitted, and only few files are uploaded, we get `array(null, <file>, null, null)` for instance. And as the data model is a collection, i guess `array(<file>)` is actually expected.